analysis_options.yaml

```diff
@@ -15,69 +15,130 @@ linter:
     - always_put_control_body_on_new_line
     - always_require_non_null_named_parameters
     - annotate_overrides
+    - avoid_double_and_int_checks
+    - avoid_dynamic_calls
     - avoid_empty_else
+    - avoid_equals_and_hash_code_on_mutable_classes
```

Comment:
Why does one want to avoid this?

Reply:
this page has the explanation: https://dart-lang.github.io/linter/lints/avoid_equals_and_hash_code_on_mutable_classes.html

basically it breaks let's say if you mutate the class while mid-sorting a list

nothing broke because of this rule, I just added to keep an eye in the future.

I am ok with removing if we think it doesn't make sense

Reply:
We can do the other way around, remove the rule it if it causes trouble.

Reply:
yeah I think that is a good rule of thumb!